src/app/public/modules/angular-tree/angular-tree-node.component.ts

```diff
@@ -0,0 +1,221 @@
+    if (value !== this._childFocusIndex) {
+      this._childFocusIndex = value;
+      if (this.focusableChildren.length > 0 && value !== undefined) {
+        this.focusableChildren[value].focus();
+      } else {
+        this.nodeContentWrapperRef.nativeElement.focus();
+      }
+    }
+  }
+
+  public get childFocusIndex(): number {
+    return this._childFocusIndex;
+  }
+
+  public set focused(value: boolean) {
+    this.tabIndex = value ? 0 : -1;
+    if (value) {
+      this.nodeContentWrapperRef.nativeElement.focus();
+    }
+  }
+
+  public set isPartiallySelected(value: boolean) {
+    if (value !== this._isPartiallySelected) {
+      this._isPartiallySelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isPartiallySelected(): boolean {
+    return this._isPartiallySelected;
+  }
+
+  public set isSelected(value: boolean) {
+    if (value !== this._isSelected) {
+      this._isSelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isSelected(): boolean {
+    return this._isSelected;
+  }
+
+  public set tabIndex(value: number) {
+    this._tabIndex = value;
+  }
+
+  public get tabIndex(): number {
+    return this._tabIndex;
+  }
+
+  @ViewChild('nodeContentWrapper')
+  public nodeContentWrapperRef: ElementRef;
+
+  private focusableChildren: HTMLElement[] = [];
+
+  private _childFocusIndex: number;
+
+  private _isPartiallySelected: boolean;
+
+  private _isSelected: boolean;
+
+  private _tabIndex: number = -1;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private adapterService: SkyAngularTreeAdapterService,
+    @Optional() private skyAngularTreeWrapper: SkyAngularTreeWrapperComponent
+  ) { }
+
+  public ngOnInit(): void {
+    if (!this.skyAngularTreeWrapper) {
+      console.warn(`<sky-angular-tree-node-wrapper> should be wrapped inside a <sky-angular-tree-wrapper> component.`);
+    }
+
+    // Because we're binding the checkbox to node's children properties, we need to manually control change detection.
+    // Here, we listen to the tree's state and force change detection in the setters if the value has changed.
+    this.node.treeModel.subscribeToState((state: ITreeState) => {
+      this.isSelected = this.node.isSelected;
+      this.isPartiallySelected = this.node.isPartiallySelected;
+
+      if (state.focusedNodeId) {
+        this.focused = state.focusedNodeId === this.node.id;
+      }
+    });
+
+    // Make the first root node tabbable.
+    if (this.node.isRoot && this.node.index === 0) {
+      this.tabIndex = 0;
+    }
+  }
+
+  public ngAfterViewInit(): void {
+    // Wait for node to render, then reset all child tabIndexes to -1.
+    setTimeout(() => {
+      this.focusableChildren = this.adapterService.getFocusableChildren(this.nodeContentWrapperRef.nativeElement);
+      this.adapterService.setTabIndexOfFocusableElems(this.nodeContentWrapperRef.nativeElement, -1);
+    }, 1000);
```

Comment:
Do we need a full 1000 (aka 1 second) here?

Reply:
I understand wanting to make sure things are ready. But a second is getting to the point where the timeout can become noticeable.

Reply:
This was a tricky one. When there's multiple trees on one page, the inner components don't get rendered to the DOM at the point `ngAfterViewInit()` runs. The inner components need to be there so these couple lines of logic can manage all the tab indexes. `setTimeout()` with the default `0ms` also didn't work. I tried a couple different ways to solve this but ended up to be the simplest and I thought it was an ok trade off because it only is impacting `tabIndex`. Its not visible to the end user, and they wouldn't know its happening unless they begin tabbing through the DOM immediately.
Do you have a suggestion on a better way to do this?

Reply:
There aren't any "ready" events you can tie into? At the very least, we'll need a comment explaining why the timeout is considerable.

Reply:
There is a `initialized` event that comes with the library, but I was finding this would fire even before the elements were painted to the screen. The time out was the best compromise I could find.

Reply:
I think I'm ok with it if we do need the full `1000ms`. I would have liked it a little better if we could get away with `100` or even `500` but in my opinion it shouldn't hold up the PR just for this.